src/net_processing.cpp

```diff
@@ -4577,12 +4822,43 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                             pto->m_tx_relay->filterInventoryKnown.insert(txid);
                         }
                     }
+
+                    // Populating local reconciliation set.
+                    if (txs_to_reconcile.size() != 0) {
+                        m_reconciliation.AddToReconSet(pto->GetId(), txs_to_reconcile);
+                    }
                 }
             }
         }
         if (!vInv.empty())
             m_connman.PushMessage(pto, msgMaker.Make(NetMsgType::INV, vInv));
 
+        //
+        // Message: reconciliation request
+        //
+        {
+            auto reconciliation_request_data = m_reconciliation.MaybeRequestReconciliation(pto->GetId());
+            if (reconciliation_request_data) {
+                const auto [local_set_size, local_q_formatted] = (*reconciliation_request_data);
+                m_connman.PushMessage(pto, msgMaker.Make(NetMsgType::REQRECON, local_set_size, local_q_formatted));
+            }
+        }
+
+        //
+        // Message: reconciliation response
+        //
+        {
+            std::vector<uint8_t> skdata;
+            bool respond = m_reconciliation.RespondToReconciliationRequest(pto->GetId(), skdata);
```

Comment:
>is MaybeRequestReconciliation going to generate a new reconciliation request ?

No, because `if (recon_state->second.m_state_init_by_us.m_phase != Phase::NONE) return std::nullopt;` takes care of that. Basically, an initiator will wait forever.